feat(cli): add profile info and -switcher to cli

PVermeer opened the pull request:
I currently have a script for automatic switching profiles via `lactd.socket`.

This however requires some dependencies like `ncat` and `jq`. Since I'm on an immutable OS where not all of these dependencies are available by default I would think that this could be done with just the LACT cli. So I did a check, and all the methods for this are already implemented so they only had to be mapped to cli.

This PR should make it a great deal easier to have script change profiles (without external deps).

Something like this is currently needed to get the current profile:
```sh
echo "{\"command\": \"list_profiles\", \"args\": {\"include_state\": true }}" | ncat -U /run/lactd.sock | jq -r .data.current_profile
```
This can now be done with:
```sh
lact cli current-profile
# Or flatpak
flatpak run io.github.ilya_zlobintsev.LACT cli current-profile
```
**Spec**
```sh
Usage: lact cli [OPTIONS] <COMMAND>

Commands:
  list-gpus        List GPUs
  info             Show GPU info
  snapshot         Generate debug snapshot
  list-profiles    List profiles
  current-profile  Current profile
  set-profile      Set profile
  help             Print this message or the help of the given subcommand(s)

Options:
  -g, --gpu-id <GPU_ID>  
  -h, --help             Print help
  -V, --version          Print version

# ===

Usage: lact cli set-profile --name <NAME> --auto-switch <AUTO_SWITCH>

Options:
  -n, --name <NAME>                
  -a, --auto-switch <AUTO_SWITCH>  [possible values: true, false]
  -h, --help                       Print help
```

**Tested with:**
- Fedora binaries
- Flatpak build

**Notes**
The `--auto-switch` argument is non-optional in the code but is optional via the socket. I think optional is better but it would require some further code changes.
https://github.com/ilya-zlobintsev/LACT/blob/e4d1f1d2669a8d2869ee7e61a491144985432220/lact-client/src/lib.rs#L154

PVermeer commented on lact-cli/src/lib.rs:
```diff
@@ -70,3 +73,29 @@ async fn snapshot(client: &DaemonClient) -> Result<()> {
 }
+
+async fn list_profiles(_: &CliArgs, client: &DaemonClient) -> Result<()> {
+    let profiles_info = client.list_profiles(true).await?;
```
Thanks, I forgot to check this.

PVermeer commented on lact-cli/src/lib.rs:
```diff
@@ -70,3 +73,29 @@ async fn snapshot(client: &DaemonClient) -> Result<()> {
+
+async fn current_profile(_: &CliArgs, client: &DaemonClient) -> Result<()> {
+    let profiles_info = client.list_profiles(true).await?;
+    if let Some(current_profile) = profiles_info.current_profile {
+        println!("{}", current_profile);
+    }
```
I agree

ilya-zlobintsev reviewed (commented):
I like this overall, just a small suggestion: it would be nice if the profile list highlighted what current profile is, as well as list the default profile (otherwise it's not very obvious that you can switch to it with `lact cli profile set Default`)

PVermeer commented:
I've updated the cli with:
```sh
$ ./lact cli profile --help
Manage profiles or show current profile

Usage: lact cli profile [COMMAND]

Commands:
  list         List profiles
  get          Current profile
  set          Set profile
  auto-switch  Manage profile auto switching
  help         Print this message or the help of the given subcommand(s)

Options:
  -h, --help  Print help

$ ./lact cli profile auto-switch --help
Manage profile auto switching

Usage: lact cli profile auto-switch [COMMAND]

Commands:
  get      Current auto switch state
  enable   Enable auto switching
  disable  Disable auto switching
  help     Print this message or the help of the given subcommand(s)

Options:
  -h, --help  Print help
```

**Changes**
- Changed to cli to it's own subcommand.
- Separated `auto switch profiles` to it's own nested subcommand because I found that it always immediately changes the profile to `Default`. It doesn't make sense to me have this when setting a profile.

**Notes**
> I like this overall, just a small suggestion: it would be nice if the profile list highlighted what current profile is, as well as list the default profile (otherwise it's not very obvious that you can switch to it with `lact cli profile set Default`)

Concerning your first comment, I've had highlighted what the current profile is in my first draft. I decided against it because imo it should be a list of the profiles without any clutter so it can easily be processed. The current profile can be request with the other command so keep it simple. If you disagree I will change it.

For the second part I agree. I've added the `Default` profile to the list.

**Bug?:**
Now, I've come across a bug in the server:
When `auto switching profiles` is enabled while setting a profile withe the cli, it will disable the auto switcher but keeps the profile `Default`. Setting the profile after a small delay sets the profile correctly.

I've committed a workaround in: 08f1767a09e8b9a6cdb58a2b3696fbc66475b7f1
Note: checking the current profile after setting it reports that it has set the new profile in this time frame, but switches back to `Default` after the function completes.

I've not investigated the issue further because it seems out of scope for this PR.

Thanks for taking the time to look at this.
Let me know if you need more changes.

ilya-zlobintsev reviewed (approved):
Tested and everything looks good. Thanks!

PVermeer commented:
Thanks for merging! I enjoyed working on this nice rust project.
